api/http/api_inventory_test.go

```diff
@@ -2315,15 +2335,15 @@ func TestApiInventoryInternalDevicesStatus(t *testing.T) {
 			ctx := contextMatcher()
 
 			switch tc.status {
-			case "accepted", "preauthorized", "pending":
+			case "accepted", "preauthorized", "pending", "noauth":
 				// Update statuses
 				inv.On("UpsertDevicesAttributes",
 					ctx,
 					tc.inputDeviceIDs,
 					deviceAttributes,
 				).Return(tc.UpdateResult, tc.inventoryErr)
 
-			case "decommissioned", "rejected":
+			case "decommissioned":
```

Comment:
Looks fine, just one question here: why are we not removing anymore the device on rejected?

Reply:
so, there are 2 things:
- we never did the removal, the test was wrong simply; if you look at the code, we delete only on decommissioned, so I just removed a mistake in test
- the recent discussion on this issue confirmed that this is what we want (keep devices around even if they're rejected, have no authsets, etc.). anyway, it was reiterated by Eystein once again

Reply:
Thanks for the clarification!

Reply:
...sorry, the discussion is under the original bug report:
https://tracker.mender.io/browse/MEN-2558

exchange starts here:
https://tracker.mender.io/browse/MEN-2558?focusedCommentId=108634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-108634